File: SetMismatch/SetMismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        n = len(nums)
        s = n*(n+1)//2
        miss = s - sum(set(nums))
        duplicate = sum(nums) + miss - s
        return [duplicate, miss]


class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        return [sum(nums) - sum(set(nums)), sum(range(1, len(nums)+1)) - sum(set(nums))]

File: SetMismatch/SetMismatch.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function (nums) {
  let map = new Map();
  let len = nums.length;
  let sum = (len * (len + 1)) / 2;
  let s = 0,
    a = 0;
  for (let i = 0; i < len; i++) {
    if (map.has(nums[i])) a = nums[i];
    else {
      map.set(nums[i], true);
      s += nums[i];
    }
  }
  return [a, sum - s];
};
